### PR TITLE
New and revised timestamp docs

### PR DESCRIPTION
**Pull request by ttmc**
I revised the BigchainDB documentation on timestamps:

* Added a new "Timestamps in BigchainDB" page in the Topic Guides.
* Fixed the docstring for the `timestamp()` function. It gets the Unix time, not UTC. The two are related, but different. In particular, UTC has leap seconds but Unix time doesn't.
* Made a suggestion on NTP daemon settings (if doing cluster node setup).
* Revised the `timestamp` explanations in the page about data models (transactions, blocks, votes)

### CODE REVIEW

File: docs/source/topic-guides/timestamps.md [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+
+> "The most common approach is to simply step the clock back by one second when the clock gets to 00:00:00 UTC. This is implemented in the Linux kernel and it is enabled by default when the clock is synchronized with NTP servers by the ntpd or chronyd daemon from the reference or chrony NTP implementations respectively."
+
+> "... There will be a backward step, but the clock will be off only for one second."
+
+We suggest that BigchainDB nodes run their NTP daemon in a mode which tells the kernel to step the system clock back by one second when a leap second occurs, rather than using one of the fancy "slewing" or "smearing" options. That way, there's only a small set of ambiguous timestamps (i.e. the ones associated with leap seconds).
+
+The result is that some timestamps can't be converted unambigously to a single UTC timestamp, but that only happens for leap seconds, and leap seconds are rare. (Only 26 had been inserted between 1972 and the end of 2015.)
+
+**So long as you avoid the leap seconds, you can convert BigchainDB timestamps (Unix time timestamps) to UTC unambiguosly using any standard conversion tool or library function.**
```
Comment (sbellem) [inline docs/source/topic-guides/timestamps.md]: unambiguosly --> unambiguously

File: docs/source/topic-guides/timestamps.md [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+
+## How the timestamp() Function Works
+
+BigchainDB has a utility function named `timestamp()` which amounts to:
+```python
+timestamp() = str(round(time.time()))
+```
+
+In other words, it calls the `time()` function in Python's `time` module, rounds that to the nearest integer, and converts the result to a string.
+
```
Comment (sbellem) [inline docs/source/topic-guides/timestamps.md]: Noteworthy: [round()](https://docs.python.org/3/library/functions.html#round)
> For the built-in types supporting round(), values are rounded to the closest multiple of 10 to the power minus ndigits; if two multiples are equally close, rounding is done toward the even choice (so, for example, both round(0.5) and round(-0.5) are 0, and round(1.5) is 2).

**Comment by sbellem** [comment]
:+1:

File: docs/source/topic-guides/timestamps.md (hunk 1 above)
Comment (ttmc) [inline docs/source/topic-guides/timestamps.md]: I'll fix that.

File: docs/source/topic-guides/timestamps.md (hunk 2 above)
Comment (ttmc) [inline docs/source/topic-guides/timestamps.md]: That's interesting! I'm not sure it's relevant though. Maybe I'll just link to Python's documentation for `round()` for those who are curious.

Comment (sbellem) [inline docs/source/topic-guides/timestamps.md]: :+1:

**Comment by r-marques** [comment]
:+1:

**Comment by ttmc** [comment]
Please don't merge this yet. I still want to clarify how to do (BigchainDB timestamp) --> UTC conversion for leap seconds, with examples.

**Pull request merged**